napari/_vispy/layers/base.py

```diff
@@ -67,6 +69,7 @@ def __init__(self, layer, node):
         self.layer.experimental_clipping_planes.events.connect(
             self._on_experimental_clipping_planes_change
         )
+        self.layer.events._overlays.connect(self._on_overlays_change)
```

Comment:
I have this TODO higher up in the `__init__`:

```
        # TODO: we try to avoid inner event connection, but this might be the only way
        #       until we figure out nested evented objects
        # self._overlays.events.connect(self.events._overlays)
```

The idea is that any addition/removal of overlays to the evented dict will fire it. This should allow us to do fancier stuff with overlays (like allowing plugins to provide their own :))